index.js

```diff
@@ -26,22 +27,26 @@ module.exports = class LockFile {
     if (this.locked) return
 
     try {
-      this.locking = this._lock()
+      this.locking = this._lock(this._wait)
       await this.locking
     } catch (err) {
       this.locking = null
       throw err
     }
   }
 
-  async _lock () {
+  async _lock (wait) {
     const fd = await open(this.filename)
 
     try {
       if (this.closed) throw new Error('Lock is closed')
-      const existing = LOCKS.get(this.filename)
-      if (existing && existing !== this) throw new Error('ELOCKED: ' + this.filename)
-      if (!fsx.tryLock(fd)) throw new Error('ELOCKED: ' + this.filename)
+      if (this._wait) {
+        await fsx.waitForLock(fd)
+      } else {
+        const existing = LOCKS.get(this.filename)
+        if (existing && existing !== this) throw new Error('ELOCKED: ' + this.filename)
```

Comment:
this check needs to be done above, prop move this into a function

Reply:
you mean before `waitForLock`?